### ISSUE TITLE
同一个程序单卡跑没问题，多卡比如export CUDA_VISIBLE_DEVICES=0,1,2，就报断错

### ISSUE THREAD

**Issue by 597477803** [PaddlePaddle/Paddle#21115]
------------------------------------------------
Device count: 3
Num train examples: 96220
Max train steps: 3006
Num warmup steps: 0
memory_optimize is deprecated. Use CompiledProgram and Executor
Theoretical memory usage in training: 5334.724 - 5588.759 MB
W1111 17:19:20.743114 159621 device_context.cc:259] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 9.2, Runtime API Version: 8.0
W1111 17:19:20.746721 159621 device_context.cc:267] device: 0, cuDNN Version: 7.0.
W1111 17:19:20.746817 159621 device_context.cc:293] WARNING: device: 0. The installed Paddle is compiled with CUDNN 7.1, but CUDNN version in your machine is 7.0, which may cause serious incompatible bug. Please recompile or reinstall Paddle with compatible CUDNN version.
Load pretraining parameters from ./lym/model//params.
I1111 17:19:22.966073 159621 parallel_executor.cc:329] The number of CUDAPlace, which is used in ParallelExecutor, is 3. And the Program will be copied 3 copies
script/run_lcqmc.sh: line 30: 159621 Segmentation fault      python -u run_classifier.py --use_cuda true --verbose true --do_train true --do_val true --do_test true --batch_size 32 --init_pretraining_params ${MODEL_PATH}/params --train_set ${TASK_DATA_PATH}/lcqmc/train.tsv --dev_set ${TASK_DATA_PATH}/lcqmc/dev.tsv --test_set ${TASK_DATA_PATH}/lcqmc/test.tsv --vocab_path config/vocab.txt --checkpoints ./checkpoints --save_steps 1000 --weight_decay 0.0 --warmup_proportion 0.0 --validation_steps 100 --epoch 3 --max_seq_len 128 --ernie_config_path config/ernie_config.json --learning_rate 2e-5 --skip_steps 10 --num_iteration_per_drop_scope 1 --num_labels 2 --random_seed 1

**Comment by 597477803** [comment]
想确认下，centos6支持多卡么？

**Comment by sandyhouse** [comment]
有更详细的报错信息吗，可以设置export GLOG_v=3试试。

**Comment by 597477803** [comment]
> 有更详细的报错信息吗，可以设置export GLOG_v=3试试。

上面贴的就是全部错误输出了

**Comment by 597477803** [comment]
> 有更详细的报错信息吗，可以设置export GLOG_v=3试试。

I1111 18:28:16.415174 153887 graph.cc:119] deal with var: word_embedding
I1111 18:28:16.415185 153887 graph.cc:119] deal with var: word_embedding_beta1_pow_acc_0
I1111 18:28:16.415191 153887 graph.cc:119] deal with var: word_embedding_beta2_pow_acc_0
I1111 18:28:16.415199 153887 graph.cc:119] deal with var: word_embedding_moment1_0
I1111 18:28:16.415203 153887 graph.cc:119] deal with var: word_embedding_moment2_0
I1111 18:28:16.483609 153887 parallel_executor.cc:329] The number of CUDAPlace, which is used in ParallelExecutor, is 3. And the Program will be copied 3 copies
I1111 18:28:16.483652 153887 parallel_executor.cc:198] not find NCCLCommunicator in scope, so recreate it!
I1111 18:28:16.483664 153887 scope.cc:164] Create variable NCCLCommunicator
I1111 18:28:16.483670 153887 parallel_executor.cc:99] nccl comm num:1, nranks:3, num_trainers:1, trainer_id:0
I1111 18:28:16.940050 153887 dynamic_loader.cc:83] Try to find library: libnccl.so from default system path.
script/run_lcqmc.sh: line 31: 153887 Segmentation fault      python -u run_classifier.py --use_cuda true --verbose true --do_train true --do_val true --do_test true --batch_size 32 --init_pretraining_params ${MODEL_PATH}/params --train_set ${TASK_DATA_PATH}/lcqmc/train.tsv --dev_set ${TASK_DATA_PATH}/lcqmc/dev.tsv --test_set ${TASK_DATA_PATH}/lcqmc/test.tsv --vocab_path config/vocab.txt --checkpoints ./checkpoints --save_steps 1000 --weight_decay 0.0 --warmup_proportion 0.0 --validation_steps 100 --epoch 3 --max_seq_len 128 --ernie_config_path config/ernie_config.json --learning_rate 2e-5 --skip_steps 10 --num_iteration_per_drop_scope 1 --num_labels 2 --random_seed 1
^C
[1]+  Exit 139                sh script/run_lcqmc.sh > 2.log 2>&1
输出了很多警告，但是还是报这个段错误

**Comment by sandyhouse** [comment]
能否提供可复现的代码，或者提供相关链接。

**Comment by 597477803** [comment]
> 能否提供可复现的代码，或者提供相关链接。
并没有自己写代码，用的是ernie里的标准程序进行多卡测试；nccl我试了几个版本，发现在目标机器上都不成功，然后但是同样的环境在另外一台机器上跑成功了，所以现在猜测，可能是目标机器的单卡GPU内存不够（只有7.5G），而那台成功的机器的单卡GPU是11G的。

**Comment by sandyhouse** [comment]
那可以通过batch size等方法试一试。另外，你的日志重定向到2.log中，里面有什么相关信息吗？ @597477803

**Comment by 597477803** [comment]
> 那可以通过batch size等方法试一试。另外，你的日志重定向到2.log中，里面有什么相关信息吗？ @597477803

batch size已经调到16了，就跟上线的日志一样，没什么有用的信息，就是Segmentation fault 就没了

**Comment by 597477803** [comment]
> 那可以通过batch size等方法试一试。另外，你的日志重定向到2.log中，里面有什么相关信息吗？ @597477803
batch size已经调到16了，就跟上面的日志一样，没什么有用的信息，就是Segmentation fault 就没了

**Comment by sandyhouse** [comment]
在11G显存的机器上无法复现吗？如果无法复现的话，可以尝试在有问题的机器上设置GLOG_v=10试试是否有有效信息；如果可以复现的话，请提供可复现的代码，我们具体复现看一下。

**Comment by 597477803** [comment]
> 在11G显存的机器上无法复现吗？如果无法复现的话，可以尝试在有问题的机器上设置GLOG_v=10试试是否有有效信息；如果可以复现的话，请提供可复现的代码，我们具体复现看一下。

**Comment by 597477803** [comment]
> 在11G显存的机器上无法复现吗？如果无法复现的话，可以尝试在有问题的机器上设置GLOG_v=10试试是否有有效信息；如果可以复现的话，请提供可复现的代码，我们具体复现看一下。

11g的机器无法复现，在7.5g的机器上试了，用script/pretrain.sh跑，发现把batch改小后能跑单卡，改成多卡还是报错，所以排除内存不够的原因，应该是其他问题

**Comment by sandyhouse** [comment]
方便设置GLOG_v=10，然后贴下输出的log吗
